### PR TITLE
move InternalCommitTrigger.Intents one level up

### PR DESCRIPTION
**Pull request by tbg**
and add EndKey to Intent (unused at this point).

this is part of #1873. Passing the `Intent`s via `EndTransaction`
is going to be how intents are resolved at the `Range` level.

### CODE REVIEW

**Comment by tamird** [comment]
LGTM. So here a misbehaving client just results in lingering intents which are cleaned up on the next read?

**Comment by tschottdorf** [comment]
yes, nothing has changed from the status quo except that the previous PR necessitates that you lie for that to happen.

**Comment by tamird** [comment]
There's a typo in the commit message, I think you meant `InternalCommitTrigger`.

**Comment by tschottdorf** [comment]
yup. Updated.

File: proto/data.proto [hunk 1]
```diff
@@ -252,10 +247,11 @@ message Lease {
       (gogoproto.customname) = "RaftNodeID", (gogoproto.casttype) = "RaftNodeID"];
 }
 
-// Intent groups a key and a transaction.
+// Intent is used to communicate the location of an intent.
```
Comment (spencerkimball) [inline proto/data.proto]: I think this should be moved to api.proto as well.

File: storage/range_command.go [hunk 2]
```diff
@@ -306,19 +306,20 @@ func (r *Range) EndTransaction(batch engine.Engine, ms *engine.MVCCStats, args p
 		return reply, err
 	}
 
+	// Resolve any explicit intents.
+	for _, intent := range args.Intents {
```
Comment (spencerkimball) [inline storage/range_command.go]: Might want a TODO here to differentiate between intents which fall within this range and otherwise.

Comment (tschottdorf) [inline storage/range_command.go]: this part is under heavy duty construction right now. See #1873.

**Pull request merged**